### PR TITLE
[Architecture] ToolsTab

### PR DESCRIPTION
**Pull request by Doomsdayrs**
### Description of the Change

Migrate ToolsTab business to using a view model

### Testing

- [x] Network Checker
- [x] Log Clearer
- [ ] Debug Mode
- [ ] Runtime Downloader
  - [ ] Launch
  - [ ] Remove 
- [x] Downloader Clearer
- [x] Skin Updater

### Related Issues

#606

### CODE REVIEW

**Comment by Doomsdayrs** [comment]
@RyanTheAllmighty Can you test the remainder tool options and finalize with a review?

File: src/main/java/com/atlauncher/gui/tabs/tools/SkinUpdaterToolPanel.java [hunk 1]
```diff
@@ -24,57 +24,43 @@
...
-    }
-
-    private void checkLaunchButtonEnabled() {
-        LAUNCH_BUTTON.setEnabled(Data.ACCOUNTS.size() != 0);
+        viewModel.onSkinUpdaterEnabledChanged(LAUNCH_BUTTON::setEnabled);
     }
 
     @Override
     public void actionPerformed(ActionEvent e) {
         Analytics.sendEvent("SkinUpdater", "Run", "Tool");
```
Comment (RyanTheAllmighty) [inline src/main/java/com/atlauncher/gui/tabs/tools/SkinUpdaterToolPanel.java:50]: It seems off to me that half of these analytics events are sent within the tool panel itself and the rest in the view model, so there's no 1 clear place to know where to see the analytics events. I think no matter where they are, they should be consistently in one or the other

**Review by RyanTheAllmighty — CHANGES_REQUESTED** [review]

File: src/main/java/com/atlauncher/gui/tabs/tools/SkinUpdaterToolPanel.java (hunk 1 above)
Comment (Doomsdayrs) [reply src/main/java/com/atlauncher/gui/tabs/tools/SkinUpdaterToolPanel.java]: Analytic events should only occur in view model.

The one purpose of the UI layer is to handle UI and updating the UI. Everything else should be delegated to the view model.

Moved calls in [9dcaa4f](https://github.com/ATLauncher/ATLauncher/pull/612/commits/9dcaa4f2c92f5d03a49b339968c205688ac731f2)

File: src/main/java/com/atlauncher/gui/tabs/tools/RuntimeDownloaderToolPanel.java [hunk 2]
```diff
@@ -17,201 +17,89 @@
...
+            );
 
             dialog.close();
         }));
 
         dialog.start();
 
-        if (dialog.getReturnValue() == null) {
-            LOG.error("Runtime downloaded failed to run!");
+        if (dialog.getReturnValue()) {
```
[the remark this answers is not in the document]
Comment (Doomsdayrs) [reply src/main/java/com/atlauncher/gui/tabs/tools/RuntimeDownloaderToolPanel.java]: Inverted in [8933f27](https://github.com/ATLauncher/ATLauncher/pull/612/commits/8933f2796890cfb4074886ab700d7b8428fd1092)

**Review by RyanTheAllmighty — APPROVED** [review]

**Review by RyanTheAllmighty — APPROVED** [review]

**Pull request merged**